### ISSUE TITLE
maven scopes

### ISSUE THREAD

**Issue by nfisher** [bazelbuild/rules_jvm_external#142]
Would it make sense to have a `scope` argument on the `maven.artifact` function for a more direct relationship to maven's scope types? The main benefit I would see is that it provides a direct mechanism to translate between Maven and Bazel without requiring people deep-diving into Bazel's dependency graph modeling upfront.

You're probably aware but the 6 scopes are:

* Compile (default).
* Provided.
* Runtime.
* Test.
* System.
* Import (not relevant to Bazel directly maybe to this tool).

Test and compile feel like they would basically be nops in terms of additional behaviour. 

Runtime, Provided, and System could probably set `neverlink=True` but might need some more thought.

Feels roughly related to #132 ignoring the maven type.

**Comment by jin** [comment]
Just to be clear, are there any functionalities around scopes that cannot be expressed using the current maven_install API?

**Comment by nfisher** [comment]
I *believe* all of them that are relevant can be modeled. I'm currently experimenting with converting a large-ish (~50) multi-module project and can feedback what I find.

**Comment by jin** [comment]
@nfisher hey, any updates with your conversion?

**Comment by nfisher** [comment]
Hi @jin apologies I’ve stopped using this module but the more I explored the less appropriate it felt to have a maven scope specified. As an example test libraries could move between a compile (when building test libraries for domain specific test tools) and test scope (standard tests).

The reason for moving away from it was that I found the speed hit to reevaluate the dep graph on a single dependency change or clean checkout outweighs the convenience that the api provides. I think I could probably live with the cost for dep changes but feels unnecessarily expensive for a clean checkout.

I’ve taken an approach that simply lists all of the transitive and direct dependencies as `maven_jar` targets and committing that.

Feel free to close this issue.
